commit 1a70200f404ae210b4f0334e3936e84f8edb6bc8
Author: Hannes Reinecke <hare@suse.de>
Date:   Mon Jun 27 11:52:07 2022 +0200

    nvmet-auth: expire authentication sessions
    
    Each authentication step is required to be completed within the
    KATO interval (or two minutes if not set). So add a workqueue function
    to reset the transaction ID and the expected next protocol step;
    this will automatically the next authentication command referring
    to the terminated authentication.
    
    Signed-off-by: Hannes Reinecke <hare@suse.de>
    Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@kernel.dk>

diff --git a/drivers/nvme/target/nvmet.h b/drivers/nvme/target/nvmet.h
index b76b2911234a..6ffeeb0a1c49 100644
--- a/drivers/nvme/target/nvmet.h
+++ b/drivers/nvme/target/nvmet.h
@@ -109,6 +109,7 @@ struct nvmet_sq {
 	u32			sqhd;
 	bool			sqhd_disabled;
 #ifdef CONFIG_NVME_TARGET_AUTH
+	struct delayed_work	auth_expired_work;
 	bool			authenticated;
 	u16			dhchap_tid;
 	u16			dhchap_status;